processing/src/main/java/org/apache/druid/query/filter/LikeDimFilter.java

```diff
@@ -363,5 +363,26 @@ public SuffixMatch getSuffixMatch()
     {
       return suffixMatch;
     }
+
+    @Override
+    public boolean equals(Object o)
+    {
+      if (this == o) {
+        return true;
+      }
+      if (o == null || getClass() != o.getClass()) {
+        return false;
+      }
+      LikeMatcher that = (LikeMatcher) o;
+      return getSuffixMatch() == that.getSuffixMatch() &&
+             Objects.equals(getPrefix(), that.getPrefix()) &&
+             Objects.equals(pattern.toString(), that.pattern.toString());
+    }
+
+    @Override
+    public int hashCode()
+    {
+      return Objects.hash(getSuffixMatch(), getPrefix(), pattern.toString());
+    }
```

Comment:
Hm, it doesn't look like we have EqualsVerifierTest for any of the `DimFilter` implementations; since this patch is only adding equals implementations for `Filter` classes, I think that should be done in a separate PR